data/helpers.d/print

```diff
@@ -29,3 +28,99 @@ ynh_no_log() {
   sudo mv ${ynh_cli_log}-move ${ynh_cli_log}
   return $?
 }
+
+# Main printer, just in case in the futur we have to change anything about that.
+#
+# [internal]
+#
+ynh_print_log () {
+  echo -e "${1}"
+}
+
+# Print a warning on stderr
+#
+# usage: ynh_print_warn "Text to print"
+# | arg: text - The text to print
+ynh_print_warn () {
+  ynh_print_log "\e[93m\e[1m[WARN]\e[0m ${1}" >&2
+}
+
+# Print a error on stderr
+#
+# usage: ynh_print_err "Text to print"
+# | arg: text - The text to print
+ynh_print_err () {
+  ynh_print_log "\e[91m\e[1m[ERR]\e[0m ${1}" >&2
+}
+
+# Execute a command and print the result as an error
+#
+# usage: ynh_exec_err command to execute
+# usage: ynh_exec_err "command to execute | following command"
+# In case of use of pipes, you have to use double quotes. Otherwise, this helper will be executed with the first command, then be send to the next pipe.
+#
+# | arg: command - command to execute
+ynh_exec_err () {
+	ynh_print_err "$(eval $@)"
+}
+
+# Execute a command and print the result as a warning
+#
+# usage: ynh_exec_warn command to execute
+# usage: ynh_exec_warn "command to execute | following command"
+# In case of use of pipes, you have to use double quotes. Otherwise, this helper will be executed with the first command, then be send to the next pipe.
+#
+# | arg: command - command to execute
+ynh_exec_warn () {
+	ynh_print_warn "$(eval $@)"
+}
+
+# Execute a command and force the result to be printed on stdout
+#
+# usage: ynh_exec_warn_less command to execute
+# usage: ynh_exec_warn_less "command to execute | following command"
+# In case of use of pipes, you have to use double quotes. Otherwise, this helper will be executed with the first command, then be send to the next pipe.
+#
+# | arg: command - command to execute
+ynh_exec_warn_less () {
```

Comment:
ynh_exec_stderr_to_log...? :slightly_smiling_face:

Reply:
ynh_exec_silent_error or something like, something clear and simple

Reply:
In my opinion we neither "warn less" or "silence errors" as errors are redirected to the standard output... Do we really need a helper for that...?

Reply:
Well, the idea is clearly to prevent any error to be send to stderr, and so, to jump to the face of the user as a "Warning".
Of course, said like that we would say, why avoiding errors to be visible by the user ? The answer is simple, because stderr is used for many messages which are not errors, but just simple warnings.
For example, wget print almost everything in stderr. Any build with make or something else will do the same thing. Node is also a big fan of stderr.
There's many contexts where we'd want a command to not use stderr but stdin.

Reply:
OK, I totally understand the use. Yet, I can't find an appropriate name (I just couldn't understand what the helper did without looking at the code), and won't people that need to do that know the "2>&1" option...?

Reply:
I'm sure many people already know that, but that the purpose of this helper, allow everyone to use that option easily without playing with the outputs of bash.
And, as I can see with many apps, many people as well seems to not know that option...

Reply:
Well, imho a user who has an understanding clear enough to recognize stderr outputs from stdin outputs, and to decide that he/she can safely lower the visibility of stderr outputs... clearly could learn about 2>&1 (especially from other apps if they are well commented :innocent:) and wouldn't need a helper that does just that.
Anyway, I don't see the both of us agree on that one, so I'll back out and let someone else speak up :wink:

Reply:
Let's get other point of view of this question then.
@frju365 @Josue-T

Reply:
Yes I think too that `ynh_exec_warn_less` is not really intuitive. Mybe `ynh_exec_to_info` or something like that, to mean clearly that all output will be written as info.

Reply:
Actually, we can't say "info" since "info" is a specific output, see https://github.com/YunoHost/yunohost/pull/383